src/customFileParser/unpackGeometry/main.ts

```diff
@@ -1,22 +1,111 @@
-import { RenderedGeometryGroups, GeometryIndexHandler } from './../sharedFileParserTypes';
+import { CompressedGeometryData, UncompressedValues } from './../sharedFileParserTypes';
 import PropertyLoader from './../PropertyLoader';
-import { renderedGeometryToAddFunction } from '../parserParameters';
+import { renderedPrimitiveToAddFunction, renderedPrimitivesPerFilePrimitive, renderedPrimitiveToGroup }
+  from '../parserParameters';
+import unpackInstancedMeshes from './InstancedMesh';
+import unpackMergedMeshes from './MergedMesh';
+import { Sector } from '../..';
 
-export default function unpackGeometry(
-  renderedGeometryGroups: RenderedGeometryGroups, geometryIndexHandlers: GeometryIndexHandler[],
-  uncompressedValues: any, groupName: string) {
-  const data = new PropertyLoader(uncompressedValues);
+export { unpackInstancedMeshes, unpackMergedMeshes };
+
+function createGroup(sector: Sector, primitiveName: string, capacity: number) {
+  const createdGroup = new renderedPrimitiveToGroup[primitiveName](capacity);
+  sector.primitiveGroups.push(createdGroup);
+  return createdGroup;
+}
+
+function findOrCreateDestinationGroup(
+  originalSector: Sector,
+  renderedPrimitiveInfo: {name: string, count: number},
+  numOfGeometries: number) {
+
+  let searchSector: Sector | undefined = originalSector;
+  let destinationGroup: any = undefined;
+
+  while ((searchSector !== undefined) && (destinationGroup === undefined)) {
 
-  geometryIndexHandlers.forEach(geometryIndexHandler => {
-    if (geometryIndexHandler.name === groupName) {
-      for (let j = 0; j < geometryIndexHandler.geometryCount; j++) {
-        data.loadData(geometryIndexHandler);
-        // @ts-ignore
-        renderedGeometryToAddFunction[geometryIndexHandler.name].call(this, renderedGeometryGroups, data);
+    searchSector.primitiveGroups.forEach(primitiveGroup => {
+      if (primitiveGroup.type === renderedPrimitiveInfo.name) {
+        if (primitiveGroup.capacity >= primitiveGroup.data.count + renderedPrimitiveInfo.count) {
+          destinationGroup = primitiveGroup;
+        }
       }
-      return;
+    });
+
+    searchSector = searchSector.parent;
+  }
+
+  if (destinationGroup) {
+    return destinationGroup;
+  } else {
+    if (originalSector.children.length === 0) {
+      const capacity = renderedPrimitiveInfo.count * numOfGeometries;
```

Comment:
I think we should use either `numberOfGeometries` everywhere, or `numGeometries` everywhere. Previous, you've had numberOf (which I like). So stick to one naming convention here?

Reply:
Done